### PR TITLE
Closing all the plots that ARMI opens

### PR DESCRIPTION
**Pull request by john-science**
## What is the change?

This PR makes sure that matplotlib plots in ARMI are closed.

## Why is the change being made?

Someone recently complained to me that during their recent ARMI Application runs, they were getting `matplotlib` warnings that they had more than 20 plots open at a time.

I believe this is the result of an old design idea that ARMI [should traffic in figure objects](https://github.com/terrapower/armi/issues/104), passing them around freely. Maybe it was a nice idea, we could discuss, but I think it leaves open the real possibility that plots aren't closed when they should be.

---

## Checklist

- [X] This PR has only [one purpose or idea](https://terrapower.github.io/armi/developer/tooling.html#one-idea-one-pr).
- [X] [Tests](https://terrapower.github.io/armi/developer/tooling.html#test-it) have been added/updated to verify any new/changed code.

- [X] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [X] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [X] The [release notes](https://terrapower.github.io/armi/developer/tooling.html#add-release-notes) have been updated if necessary.
- [X] The [documentation](https://terrapower.github.io/armi/developer/tooling.html#document-it) is still up-to-date in the `doc` folder.
- [X] The dependencies are still up-to-date in `pyproject.toml`.

### CODE REVIEW

**Comment by john-science** [comment]
I think a lot of ARMI PRs are failing right now because the unit tests aren't cleaning up after themselves.

In particular, after all the tests are run, a file named `armiRun.h5` appears to be left in the base dir of the working area.

This should be remedied.

**Comment by opotowsky** [comment]
> I think a lot of ARMI PRs are failing right now because the unit tests aren't cleaning up after themselves.
> 
> In particular, after all the tests are run, a file named `armiRun.h5` appears to be left in the base dir of the working area.
> 
> This should be remedied.

Interesting that it looks like it's only happening on windows and not linux

**Comment by john-science** [comment]
So, matplotlib released a new version to PyPI.org today.  This removed (or changed?) the old color map logic we are using:

https://github.com/terrapower/armi/blob/872e2dac316d937d06a9d3a7a314c48075204a2b/armi/utils/reportPlotting.py#L732


You will see `get_cmap()` in [matplotlib==3.7.5](https://matplotlib.org/3.7.5/api/cm_api.html), but not in [matplotlib==3.8.4](https://matplotlib.org/3.8.4/api/cm_api.html).

**Review by opotowsky — APPROVED** [review]

File: armi/utils/plotting.py [hunk 1]
```diff
@@ -404,13 +404,12 @@ def plotFaceMap(
             )
     elif referencesToKeep:
         # Don't show yet, since it will be updated.
+        plt.close(fig)
```
Comment (onufer) [inline armi/utils/plotting.py:407]: please Delete this one. it cant close with `plotDepthMap`

Comment (opotowsky) [reply armi/utils/plotting.py:407]: ope! @john-science I guess we didn't test this internally

**Pull request merged**